Feature/webpack config tailwindcss

notjb opened the pull request:
Added
---

- [x] Add `@studiometa/webpack-config` to the template
- [x] Add tailwindcss to the template & the corresponding scss
- [x] Add the `postcss.config.js`
- [x] Add a base `tailwind.config.js`

Deleted
---

- [x] Remove the gulp logic
- [x] Remove the Husky pre-commit-hook 

🚀

Commit:
Add @studiometa/webpack-config and tailwindcss

Guillaume-meta commented:
Hello,

Je vois qu’il y a yoast, on testerait pas Rank math comme Ludo l’indiquait ?

A+

> Le 29 mai 2020 à 22:13, notjb <notifications@github.com> a écrit :
> 
> ﻿
> Added
> 
>  Add @studiometa/webpack-config to the template
>  Add tailwindcss to the template & the corresponding scss
>  Add the postcss.config.js
>  Add a base tailwind.config.js
> Deleted
> 
>  Remove the gulp logic
>  Remove the pre-commit-hook
> 🚀
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/studiometa/create-wordpress-project/pull/13
> 
> Commit Summary
> 
> Merge tag '1.0.0-alpha.6' into develop
> Reduce the number of questions to setup a project
> Update the templates dependencies
> Add GitLab CI configuration
> Fix linting tasks
> Fix the PHPCS script
> Add Wordfence plugin choice to sao features
> Add classic-editor plugin to sao choices
> Add Yoast SEO plugin to sao choices
> Merge branch 'feature/update' into develop
> Merge branch 'feature/add-plugins-to-sao-choices' into develop
> Add @studiometa/webpack-config and tailwindcss
> File Changes
> 
> M package-lock.json (73)
> M package.json (3)
> M readme.md (2)
> M saofile.js (74)
> M template/.eslintrc.js (22)
> A template/.gitlab-ci.yml (121)
> A template/.nvmrc (1)
> M template/_gitignore (2)
> M template/build/config.js (10)
> M template/composer.json (12)
> D template/gulpfile.js (47)
> A template/meta.config.js (11)
> M template/package.json (53)
> A template/postcss.config.js (26)
> M template/readme.md (12)
> A template/tailwind.config.js (25)
> R template/web/wp-content/themes/<%= slug %>/404.php (0)
> R template/web/wp-content/themes/<%= slug %>/archive.php (0)
> R template/web/wp-content/themes/<%= slug %>/footer.php (0)
> R template/web/wp-content/themes/<%= slug %>/front-page.php (0)
> R template/web/wp-content/themes/<%= slug %>/functions.php (0)
> R template/web/wp-content/themes/<%= slug %>/header.php (0)
> R template/web/wp-content/themes/<%= slug %>/index.php (0)
> R template/web/wp-content/themes/<%= slug %>/page.php (0)
> R template/web/wp-content/themes/<%= slug %>/search.php (0)
> R template/web/wp-content/themes/<%= slug %>/sidebar.php (0)
> R template/web/wp-content/themes/<%= slug %>/single.php (0)
> A template/web/wp-content/themes/<%= slug %>/src/css/_config.scss (9)
> A template/web/wp-content/themes/<%= slug %>/src/css/app.scss (30)
> R template/web/wp-content/themes/<%= slug %>/src/js/app.js (0)
> R template/web/wp-content/themes/<%= slug %>/static/fonts/.gitkeep (0)
> R template/web/wp-content/themes/<%= slug %>/static/img/.gitkeep (0)
> R template/web/wp-content/themes/<%= slug %>/static/svg/.gitkeep (0)
> R template/web/wp-content/themes/<%= slug %>/style.css (4)
> R template/web/wp-content/themes/<%= slug %>/templates/components/footer.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/components/header.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/components/menu.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/components/pagination.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/components/tease-post.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/components/tease.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/layouts/base.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/404.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/archive.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/front-page.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/index.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/page-plugin.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/page.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/search.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/single-password.twig (0)
> R template/web/wp-content/themes/<%= slug %>/templates/pages/single.twig (0)
> D template/web/wp-content/themes/<%= themeSlug %>/src/js/pages/index.js (0)
> D template/web/wp-content/themes/<%= themeSlug %>/src/js/utils/_debug.js (0)
> D template/web/wp-content/themes/<%= themeSlug %>/src/scss/_config.scss (5)
> D template/web/wp-content/themes/<%= themeSlug %>/src/scss/app.scss (11)
> Patch Links:
> 
> https://github.com/studiometa/create-wordpress-project/pull/13.patch
> https://github.com/studiometa/create-wordpress-project/pull/13.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

perruche commented on template/readme.md:
```diff
@@ -8,12 +8,20 @@
 ### SCSS
 
+Styles are managed with postCSS and tailwind css.
+The `.scss` files are stored in `/web/wp-content/themes/<%= slug %>/css`.
```
Path is missing the `src` folder

perruche commented on template/tailwind.config.js:
```diff
@@ -0,0 +1,25 @@
+    content: [
+      './web/wp-content/themes/<%= name %>/src/js/**/*.js',
+      './web/wp-content/themes/<%= name %>/src/js/**/*.vue',
+      './web/wp-content/themes/<%= name %>/**/*.tpl',
```
I think we can remove `*.tpl` from this list, but maybe we should add `*.php` files ?

perruche replied:
I think it should be `<%= slug %>` also in this path no ?

notjb replied:
Not sure if adding php files is useful since classes should be used in the view (twig) ?

perruche replied:
yes, probably better to leave it like this, and add php files if a project need it

perruche commented on template/web/wp-content/themes/<%= slug %>/src/css/_config.scss:
```diff
@@ -0,0 +1,9 @@
+\*============================================================================*/
+
+$img-dir: '/static/img/';
+$svg-dir: '/static/svg/';
```
`static` folder is in the theme directory right ? Does this work ? shouldn't it be `wp-content/theme/<%= slug %>/static/svg`

perruche reviewed (changes requested):
Good work !

Commit:
Resolve PR#13 and update some paths

- Removes .tpl from config files,
- Update static assets paths

notjb commented:
@Guillaume-meta as of today the team as not stated about using YOAST or Math Rank.
It should be added to a dedicated issue.

perruche commented:
>  It should be added to a dedicated issue.

Should we discuss this on the next dev meeting ? @titouanmathis

titouanmathis commented on template/package.json:
```diff
@@ -1,40 +1,39 @@
-    "dotenv": "^8.2.0",
+    "@studiometa/stylelint-config": "^2.0.0",
+    "@studiometa/tailwind-config": "^1.0.0-alpha.9",
+    "@studiometa/webpack-config": "^2.0.0-alpha.6",
```
You can update the `@studiometa/webpack-config` to its latest version.

titouanmathis commented on template/postcss.config.js:
```diff
@@ -0,0 +1,26 @@
+    require('postcss-import'),
+    require('tailwindcss'),
+    require('postcss-nested'),
+    require('postcss-custom-properties'),
```
We use SCSS files which are compile with the `sass-loader`, so the `postcss-import` and `postcss-nested` PostCSS plugins are not required.

titouanmathis commented on template/postcss.config.js:
```diff
@@ -0,0 +1,26 @@
+if (process.env.NODE_ENV === 'production') {
+  postcssConfig.plugins = [
+    ...postcssConfig.plugins,
+    require('@fullhuman/postcss-purgecss')({
```
The latest version of Tailwind CSS comes with Purge CSS, we can remove from the PostCSS configuration.

titouanmathis commented on template/tailwind.config.js:
```diff
@@ -0,0 +1,24 @@
+  theme: {},
+  purge: {
+    // Learn more on https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css
+    enabled: process.env.NODE_ENV === 'production',
```
This is the default value, we do not need to specify it.

notjb replied:
I'm not sure to understand your feedback, did you mean that also for line 17 ?

titouanmathis replied:
From the doc:

> Now whenever you compile your CSS with NODE_ENV set to production, Tailwind will automatically purge unused styles from your CSS.

The default behaviour of Tailwind CSS with Purge CSS is to enable it only when `NODE_ENV` is set to `production`. So the configuration of `enabled` (line 17) is redundant with the default one and therefore does not need to be specified.

titouanmathis commented on template/tailwind.config.js:
```diff
@@ -0,0 +1,24 @@
+const merge = require('lodash.merge');
+const plugin = require('tailwindcss/plugin');
```
`plugin` is not used, it can be removed.

titouanmathis reviewed (changes requested):
Some minor fixes need to be done, but well done!  👍

titouanmathis commented:
> > It should be added to a dedicated issue.
> 
> Should we discuss this on the next dev meeting ? @titouanmathis

I think someone should test it before, but I will keep this in mind 😉

Commit:
Remove useless packages

Remove purgeCSS logic from the config.postcss.js
as its managed by tailwindcss itself

Commit:
Resolve conflicts and remove useless test for purgeCSS since tailwind handles them

titouanmathis reviewed (approved):
👍
